runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -1251,7 +1262,7 @@ private void setAndVerifyOutputTimestamp() {
                       : PeriodFormat.getDefault().print(fn.getAllowedTimestampSkew().toPeriod()),
                   BoundedWindow.TIMESTAMP_MAX_VALUE));
         }
-      } else if (TimeDomain.EVENT_TIME.equals(spec.getTimeDomain())) {
+      } else if (!noOutputTimestamp && TimeDomain.EVENT_TIME.equals(spec.getTimeDomain())) {
```

Comment:
Seems if the user called `withNoOutputTimestamp()` on an event time timer, the `else` branch will be executed. Is this expected? I don't see any code preventing the user from doing so. I guess the answer is yes because if `withNoOutputTimestamp()` is called then the outputTimestamp value doesn't matter as we don't allow output from this timer. If so then the comment should also be updated.

Reply:
This was a bug - good catch.